mitmproxy/addons/next_layer.py

```diff
@@ -220,10 +220,18 @@ def _ignore_connection(
             client_hello := self._get_client_hello(context, data_client)
         ) and client_hello.sni:
             hostnames.append(client_hello.sni)
-
+        # If the client data is not a TLS record, try to extract the domain from the HTTP request
+        else:
+            if isinstance(data_client, bytes):
+                host = self._extract_host(data_client)
+                if host:
+                    hostnames.append(host)
```

Comment:
```suggestion
        elif host := self._extract_http1_host_header(data_client):
            hostnames.append(host)
```

`isinstance(data_client, bytes)` should always hold based on the functions type signature and the rest can be shortened a bit :)